components/match2/wikis/halo/match_summary.lua

```diff
@@ -6,130 +6,87 @@
 -- Please see https://github.com/Liquipedia/Lua-Modules to contribute
 --
 
+local Class = require('Module:Class')
+local Flags = require('Module:Flags')
+local Json = require('Module:Json')
 local Logic = require('Module:Logic')
 local Lua = require('Module:Lua')
 local MapModes = require('Module:MapModes')
-local OpponentDisplay = require('Module:OpponentDisplay')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
 local VodLink = require('Module:VodLink')
 
 local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper', {requireDevIfEnabled = true})
+local MatchSummary = Lua.import('Module:MatchSummary/Base', {requireDevIfEnabled = true})
 local MatchGroupUtil = Lua.import('Module:MatchGroup/Util', {requireDevIfEnabled = true})
 
-local htmlCreate = mw.html.create
-
-local _TBD_ICON = mw.ext.TeamTemplate.teamicon('tbd')
+local _EPOCH_TIME = '1970-01-01 00:00:00'
+local _EPOCH_TIME_EXTENDED = '1970-01-01T00:00:00+00:00'
 
-local CustomMatchSummary = {}
+local htmlCreate = mw.html.create
 
 local _GREEN_CHECK = '<i class="fa fa-check forest-green-text" style="width: 14px; text-align: center" ></i>'
 local _NO_CHECK = '[[File:NoCheck.png|link=]]'
+local _ICONS = {
+	check = _GREEN_CHECK,
+}
 
-function CustomMatchSummary.getByMatchId(args)
-	local match = MatchGroupUtil.fetchMatchForBracketDisplay(args.bracketId, args.matchId)
+local _LINK_DATA = {
+	vod = {icon = 'File:VOD Icon.png', text = 'Watch VOD'},
+	preview = {icon = 'File:Preview Icon32.png', text = 'Preview'},
+	lrthread = {icon = 'File:LiveReport32.png', text = 'LiveReport.png'},
+	esl = {
+		icon = 'File:ESL_2019_icon_lightmode.png',
+		iconDark = 'File:ESL_2019_icon_darkmode.png',
+		text = 'Match page on ESL'
+	},
+	faceit = {icon = 'File:FACEIT-icon.png', text = 'Match page on FACEIT'},
+	halodatahive = {icon = 'File:Halo Data Hive allmode.png',text = 'Match page on Halo Data Hive'},
+}
 
-	local wrapper = htmlCreate('div')
-		:addClass('brkts-popup')
-		:css('flex-wrap', 'unset') -- temporary workaround
-
-	local function renderOpponent(opponentIndex)
-		return OpponentDisplay.BlockOpponent{
-			flip = opponentIndex == 1,
-			opponent = match.opponents[opponentIndex],
-			overflow = 'ellipsis',
-			teamStyle = 'short',
-		}
+-- Custom Caster Class
```

Comment:
if we add this to lots of wikis imo move to MatchSummary/Base (commons)

Reply:
the placement is not relevant for the class itself
it would be a class provided by commons that CAN be used
just saying if lots of wikis use it we do not need to have the code for the class (not the usage of it) several times

as for the different  behavior: so far i mostly have seen the same being implemented 
if some wikis would want different behavior they can of course still not use the class provided via commons

Reply:
Yeah something to consider for the future if more wiki wants to adopt it